pennylane/ops/qubit/observables.py

```diff
@@ -116,7 +133,7 @@ def diagonalizing_gates(self):
         Returns:
             list: list containing the gates diagonalizing the Hermitian observable
         """
-        return [QubitUnitary(self.eigendecomposition["eigvec"].conj().T, wires=list(self.wires))]
+        return self.compute_diagonalizing_gates(self.eigendecomposition["eigvec"], self.wires)
```

Comment:
This is a nice example where we would overwrite the default signature...

Reply:
@mariaschuld could this be simplified even further by using the `hyperparameters` functionality?

Reply:
I would not be such a fan of this. The hyperparameters should not just be properties like the eigenvalues, they should be all external information needed to uniquely define the operation. I think that they will usually be given in the init signature in fact!

Reply:
You've convinced me, that makes a lot of sense!